src/data/writeToStore.ts

```
@@ -195,6 +197,31 @@ export function writeSelectionSetToStore({
   return store;
 }
 
+// Checks if the id given is an id that was generated by Apollo
+// rather than by dataIdFromObject.
+function isGeneratedId(id: string): boolean {
+  return (id[0] === '$');
+}
+
+function mergeWithGenerated(generatedKey: string, realKey: string, cache: NormalizedCache) {
+  const generated = cache[generatedKey];
+  const real = cache[realKey];
+
+  if (!isObject(generated) || !isObject(real)) {
+    return;
```

I feel that this should throw an error rather than silently failing, since this condition should never happen.